scripts/update-chart.sh

```diff
@@ -0,0 +1,258 @@
+GITHUB_ORG=${GITHUB_ORG:-$DEFAULT_GITHUB_ORG}
+
+DEFAULT_GITHUB_REPO="ack-parent-chart"
+GITHUB_REPO=${GITHUB_REPO:-$DEFAULT_GITHUB_REPO}
+
+DEFAULT_COMMIT_TARGET_BRANCH="main"
+COMMIT_TARGET_BRANCH=${COMMIT_TARGET_BRANCH:-$DEFAULT_COMMIT_TARGET_BRANCH}
+
+# DEPENDENCY_DIFF is an enumeration for storing how the chart depedencies have
+# changed between the last and current versions of the parent chart. Storing the
+# values as integers allows the script to compare values using mathematical
+# operators.
+DEPENDENCY_DIFF_PATCH="0"
+DEPENDENCY_DIFF_MINOR="1"
+DEPENDENCY_DIFF_MAJOR="2"
+
+source "$TEST_INFRA_LIB_DIR/common.sh"
+
+_upgrade_dependency_and_chart_versions() {
+    local parent_chart_diff="$DEPENDENCY_DIFF_PATCH"
+
+    pushd "$WORKSPACE_DIR" >/dev/null
+        local controller_names
+        controller_names=$(find . -maxdepth 1 -name "*-controller" -type d | cut -d"/" -f2)
+    popd >/dev/null
+
+    for controller_name in $controller_names; do
+        local service_name
+        service_name=$(echo "$controller_name"| sed 's/-controller$//g')
+        local controller_dir="$WORKSPACE_DIR/$controller_name"
+
+        local controller_chart="$controller_dir/helm/Chart.yaml"
+        if [[ ! -f "$controller_chart" ]]; then
+            continue
+        fi
+
+        # Determine if the chart is in GA
+        chart_major_version="$(yq '.version | split(".") | .[0] | sub("v(\d+)", "$1")' "$controller_chart")"
+        [[ "$chart_major_version" == "0" ]] && continue
+
+        chart_name="$(yq '.name' "$controller_chart")"
+        chart_version="$(yq '.version' "$controller_chart")"
+
+        local existing_version
+        existing_version="$(_get_chart_dependency_version "$chart_name")"
+        if [[ "$existing_version" == "" ]]; then
+            echo -e "Adding $chart_name as a new dependency \t $chart_version"
+
+            _add_chart_dependency "$chart_name" "$chart_version" "$service_name"
+            _add_chart_values_section "$service_name"
+
+            # For new chart dependencies, upgrade the minor version
+            parent_chart_diff="$DEPENDENCY_DIFF_MINOR"
+
+        elif [[ "$existing_version" != "$chart_version" ]]; then
+            echo -e "Upgrading $chart_name \t $chart_version"
+
+            _upgrade_chart_dependency "$chart_name" "$chart_version" "$service_name"
+
+            # Determine the difference in semver
+            local semver_diff
+            semver_diff="$(_get_semver_diff "$existing_version" "$chart_version")"
+            if [[ "$semver_diff" -gt "$parent_chart_diff" ]]; then
+                parent_chart_diff="$semver_diff"
+            fi
+        fi
+    done
+    
+    local current_chart_version
+    local new_chart_version
+    current_chart_version="$(yq '.version' "$PARENT_CHART_CONFIG")"
+    new_chart_version="$(_increment_chart_version "$current_chart_version" "$parent_chart_diff")"
+
+    echo "Updating chart from version $current_chart_version to $new_chart_version"
```

Comment:
Consider saying "Updating ack-chart from version...." here to avoid ambiguity (lots of mentions of charts going on)